### PR TITLE
docs: add PHPDoc to BaseBuilder

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add PHPDoc comments

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseBuilder.php [hunk 1]
```diff
@@ -86,8 +86,10 @@ class BaseBuilder
 
     /**
      * QB keys
+     * list of column names.
      *
      * @var array
+     * @phpstan-var list<string>
```
Comment (MGatner) [inline system/Database/BaseBuilder.php:92]: Just for my own edification... what is the benefit of using the vendor-specific annotation? I know Psalm and some IDEs understand `@var string[]` - won't they miss out on this nuance?

File: system/Database/BaseBuilder.php [hunk 2]
```diff
@@ -2417,6 +2424,11 @@ public function updateBatch($set = null, $constraints = null, int $batchSize = 1
 
     /**
      * Generates a platform-specific batch update string from the supplied data
+     *
+     * @used-by batchExecute
+     *
+     * @param array $keys   QBKeys
+     * @param array $values QBSet
```
Comment (MGatner) [inline system/Database/BaseBuilder.php:2431]: Can these get any more specific? Let's probably are all strings, right?

Comment (sclubricants) [reply system/Database/BaseBuilder.php:2431]: $values can be RawSql

File: system/Database/BaseBuilder.php (hunk 1 above)
Comment (kenjis) [reply system/Database/BaseBuilder.php:92]: `string[]` means `array<string>`. It does not care about keys.
https://psalm.dev/docs/annotating_code/type_syntax/array_types/#phpdoc-syntax

`list` means true list that the key starts with 0, and goes 1,2,3,...

File: system/Database/BaseBuilder.php (hunk 2 above)
Comment (kenjis) [reply system/Database/BaseBuilder.php:2431]: @sclubricants But it is `array` typed.
`protected function _updateBatch(string $table, array $keys, array $values): string`

Comment (kenjis) [reply system/Database/BaseBuilder.php:2431]: $QBKeys: `list<string>`
$QBSet: `array<string, string>|list<list<string|int>>`, but in this case `list<list<string|int>>` only.

Comment (kenjis) [reply system/Database/BaseBuilder.php]: Updated.

**Review by MGatner — APPROVED** [review]

File: system/Database/BaseBuilder.php (hunk 1 above)
Comment (paulbalandan) [reply system/Database/BaseBuilder.php:92]: VS Code does not support it yet.

**Pull request merged**